test/lib/markdown_parser_test.exs

```diff
@@ -49,4 +50,12 @@ defmodule HonGraffitiPhoenix.MarkupParserTest do
     assert hd(MarkupParser.parse("^*")).color == "white"
   end
 
+  test "it can handle multiple carets" do
```

Comment:
I think a more descriptive test name would be better, e.g.
```Elixir
test "groups multiple sequential carets into one body, except the last one" do
```

Also, what happens when the final caret is followed by an invalid color code? Such as

```Elixir
"We shouldn't freak over ^^^^^j many carets"
# or
"We shouldn't freak over ^^^^^1e2 many carets"
# or
"We shouldn't freak over ^^^^^ many carets"
```